src/pool/srv_util.c

```diff
@@ -371,3 +371,86 @@ ds_pool_map_tgts_update(struct pool_map *map, struct pool_target_id_list *tgts,
 	return 0;
 }
 
+#define SWAP_RANKS(ranks, i, j)					\
+	do {							\
+		d_rank_t r = ranks->rl_ranks[i];		\
+								\
+		ranks->rl_ranks[i] = ranks->rl_ranks[j];	\
+		ranks->rl_ranks[j] = r;				\
+	} while (0)
+
+/*
+ * Find failed ranks in `replicas` and copy to a new list `failed`
+ * Replace the failed ranks with ranks which are up and running.
+ * `alt` points to the subset of `replicas` containing replacements.
+ */
+int
+ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
+			      d_rank_list_t *failed, d_rank_list_t *alt)
+{
+	struct pool_domain	*domains = NULL;
```

Comment:
Although the type is pool_domain, I suspect these are actually nodes (otherwise they don't have ranks).

Reply:
Sure.. The `pool_map_find_nodes` call ensures only nodes are returned. Are you proposing a change here?

Reply:
"nodes" might be a more appropriate name.